tests/cxx/builtins_new_included_cxx14.cc

```diff
@@ -0,0 +1,31 @@
+//===--- builtins_new_included_cxx14.cc - test input file for iwyu --------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+// Test that iwyu suggests the include for <new> be removed if only
+// built-in functions are used, C++14 edition (i.e. with -fsized-deallocation).
+
+#include <new>
+
+void foo() {
+  char* ch = new char;
```

Comment:
It would be nice if this file demonstrated the use of sized deallocation.

Reply:
Hmm, I thought the compiler does so transparently by replacing the call site with the sized variant of `operator delete`.

FWIW, I observed that if I reverted my patch to `iwyu_ast_util.cc` and ran just the new test, it fails as IWYU sees the two-arg variant of `operator delete` with a non-const second argument (i.e. the `std::size_t`) and retains the recommendation to include `<new>`.

Reply:
@adembo Oh, nice! I'm not sure what my perception of sized deallocation was, but that makes sense.

Hmm, OK... But this introduces another annoying thing, I guess -- IWYU's detection of default-new-or-delete now depends on language mode. So if you run IWYU with `-fsized-deallocation`, it will remove an include that's required for C++11 builds (assuming for example a portable library for C++11 to C++14).

I guess the answer to that is to always run IWYU with the lowest supported language version. That should go in our docs at some point.

I've convinced myself that this is the right solution, thanks for listening :)

Reply:
Yeah it's not perfect, but I'm honestly not sure how IWYU can avoid stuff like that writ large. For example, if your portable library built with one header on one architecture and a second on another, you may end up with different IWYU results depending on which architecture you run it.

This is actually a problem we're having with IWYU in our large codebase, and we work around it via prolific use of `// IWYU pragma: keep`.